externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/SparkSQLOperationManager.scala

```diff
@@ -59,6 +61,11 @@ class SparkSQLOperationManager private (name: String) extends OperationManager(n
       queryTimeout: Long): Operation = {
     val spark = getSparkSession(session.handle)
     val operation = new ExecuteStatement(spark, session, statement, runAsync, queryTimeout)
+
+    val sparkSQLOperationListener = new SparkSQLOperationListener(operation)
+    spark.sparkContext.addSparkListener(sparkSQLOperationListener)
+
+    addListener(operation.getHandle.identifier.toString, sparkSQLOperationListener, spark)
```

Comment:
this can be a SparkContext level listener, can be added once when initializing SparkContext